persist-cli/src/main/java/io/ballerina/persist/utils/SqlScriptGenerationUtils.java

```diff
@@ -0,0 +1,352 @@
+    private static final String SET_NULL = "persist:SET_NULL";
+    private static final String NO_ACTION = "persist:NO_ACTION";
+    private static final String RESTRICT_SYNTAX = " RESTRICT";
+    private static final String CASCADE_SYNTAX = " CASCADE";
+    private static final String NO_ACTION_SYNTAX = " NO ACTION";
+    private static final String SET_NULL_SYNTAX = " SET NULL";
+    private static final String SET_DEFAULT_SYNTAX = " SET DEFAULT";
+
+    private SqlScriptGenerationUtils(){}
+
+    public static String generateSqlScript(ArrayList<Entity> entityArray) throws BalException {
+        HashMap<String, List<String>> referenceTables = new HashMap<>();
+        List<String> tableNamesInScript = new ArrayList<>();
+        String sqlScript = "";
+        for (Entity entity : entityArray) {
+            String tableName = entity.getTableName();
+            String tableScript = generateDropTableQuery(tableName) + generateTableQuery(tableName, entityArray, entity,
+                    referenceTables);
+            sqlScript = updateSqlScript(sqlScript, tableScript, entity.getTableName(), referenceTables,
+                    tableNamesInScript);
+
+        }
+        return sqlScript;
+    }
+
+    public static void addScriptToFile(String sqlScript, Path filePath) {
+        Path path = Paths.get(String.valueOf(filePath), PersistToolsConstants.FILE_NAME);
+        try {
+            Files.deleteIfExists(path);
+            Files.createFile(path);
+            Files.writeString(path, sqlScript);
+        } catch (IOException e) {
+            PrintStream errStream = System.err;
+            errStream.println("Error while updating the SQL script file (persist_db_push.sql) in the project " +
+                    "persist directory: " + e.getMessage());
+        }
+    }
+
+    private static String generateDropTableQuery(String tableName) {
+        return MessageFormat.format("DROP TABLE IF EXISTS {0};", tableName);
+    }
+
+    private static String generateTableQuery(String tableName, ArrayList<Entity> entityArray, Entity entity,
+                                             HashMap<String, List<String>> referenceTables) {
+        return MessageFormat.format("{1}CREATE TABLE {0} (", tableName, NEW_LINE) +
+                generateFieldsQuery(entity.getTableName(), Arrays.asList(entity.getKeys()),
+                        entity.getUniqueConstraints(), referenceTables, entity, entityArray);
+    }
+
+    private static String generateFieldsQuery(String tableName,
+                                              List<String> primaryKeys, List<List<String>> uniqueConstraints,
+                                              HashMap<String, List<String>> referenceTables,
+                                              Entity entity, ArrayList<Entity> entityArray) {
+        StringBuilder sqlScript = new StringBuilder();
+        String end = NEW_LINE + ");";
+        end = createColumnsScript(entity, end, sqlScript);
```

Comment:
But if any field supports auto-generated, this start value needs to be added to the end of the table script. If this will be separated to another function, we need to check all the fields again. I think we can create this script under this function as this is also a part of the filed script. WDYT?

I updated that variable name now.

Reply:
Then this function should only return, whether auto increment is present, not update the script on behalf of the calling function

Reply:
Updated in [48e40cb](https://github.com/ballerina-platform/persist-tools/pull/60/commits/48e40cb6c629f00934b97019c3b09567cd9e391f)